components/eam/src/physics/cam/convect_deep.F90

```diff
@@ -1,428 +1,331 @@
-   real(r8), pointer, dimension(:,:) :: lambdadpcu   ! Droplet size distribution shape parameter for radiation
-   real(r8), pointer, dimension(:,:) :: qi           ! wg grid slice of cloud ice.
-
-
-   real(r8), pointer, dimension(:,:) :: evapcdp   ! Evaporation of deep convective precipitation
-
-   real(r8), pointer :: pblh(:)                ! Planetary boundary layer height
-   real(r8), pointer :: tpert(:)               ! Thermal temperature excess 
-
-   ! Temperature tendency from deep convection (pbuf pointer).
-   real(r8), pointer, dimension(:,:) :: ttend_dp
-
-   real(r8) zero(pcols, pver)
-
-   integer :: jctop(pcols), jcbot(pcols)
-   integer i, k
-
+   !----------------------------------------------------------------------------
+   ! Arguments
+   type(physics_state),             intent(in ) :: state    ! Physics state variables
+   type(physics_ptend),             intent(out) :: ptend    ! individual parameterization tendencies
+   type(physics_buffer_desc),       pointer     :: pbuf(:)  ! physics buffer
+   real(r8),                        intent(in)  :: ztodt    ! 2 delta t (model time increment)
+   real(r8), dimension(pcols),      intent(in)  :: landfrac ! Land fraction
+   real(r8), dimension(pcols,pverp),intent(out) :: mcon     ! Convective mass flux--m sub c
+   real(r8), dimension(pcols,pver), intent(out) :: dlf      ! scattered detraining cld h2o tend
+   real(r8), dimension(pcols,pverp),intent(out) :: pflx     ! scattered precip flux at each level
+   real(r8), dimension(pcols,pver), intent(out) :: cme      ! mass tendency due to condensation - evaporation
+   real(r8), dimension(pcols,pver), intent(out) :: zdu      ! detraining mass flux
+   real(r8), dimension(pcols),      intent(out) :: rliq     ! reserved liq (not yet in cldliq) for energy integrals
+   real(r8), dimension(pcols),      intent(out) :: rice     ! reserved ice (not yet in cldice) for energy integrals
+   real(r8), dimension(pcols,pver), intent(out) :: mu       ! upward cloud mass flux               [?]
+   real(r8), dimension(pcols,pver), intent(out) :: eu       ! entrainment in updraft               [?]
+   real(r8), dimension(pcols,pver), intent(out) :: du       ! detrainment in updraft               [?]
+   real(r8), dimension(pcols,pver), intent(out) :: md       ! downward cloud mass flux             [?]
+   real(r8), dimension(pcols,pver), intent(out) :: ed       ! entrainment in downdraft             [?]
+   real(r8), dimension(pcols,pver), intent(out) :: dp       ! layer thickness between interfaces   [mb]
+   real(r8), dimension(pcols),      intent(out) :: dsubcld  ! layer thickness between lcl and maxi [mb]
+   integer,  dimension(pcols),      intent(out) :: jt       ! top level index of deep convection
+   integer,  dimension(pcols),      intent(out) :: maxg     ! gathered values of maxi
+   integer,  dimension(pcols),      intent(out) :: ideep    ! flag to indicate active column for gathering
```

Comment:
similar discrepancy with `maxg` and `ideep`

Reply:
see explanation above